tests/test_specific_rules.py

```diff
@@ -51,7 +51,10 @@ def test_os_and_platform_in_query(self):
             err_msg = f"{self.rule_str(rule)} missing required field for endpoint rule"
             if "host.os.type" not in fields:
                 # Exception for Forwarded Events which contain Windows-only fields.
-                if rule.path.parent.name == "windows" and not any(field.startswith("winlog.") for field in fields):
+                if rule.path.parent.name == "windows":
+                    if not any(field.startswith("winlog.") for field in fields):
+                        self.assertIn("host.os.type", fields, err_msg)
+                else:
```

Comment:
nit: 

By the time we get here, we've already checked `"host.os.type" not in fields:`, so we probably dont need to `assertIn` again. Something like `self.assertTrue(any(field.startswith("winlog.") for field in fields), "some more appropriate message about winlog and host.os.type"` might be better.

Then in the else, `self.fail(err_msg)`

Reply:
I think you're reading it wrong. If winlog fields are present, the check is abandoned. 

I initially had it collapsed more but it was much more readable this way.

Reply:
I mean compare line 52 & 56 for example or 52 & 58.